zephyr/src/supp_api.c

```diff
@@ -371,7 +371,8 @@ int zephyr_supp_status(const struct device *dev,
 			status->ssid_len = ssid_len;
 			status->iface_mode = ssid->mode;
 			/* TODO: Derive this based on association IEs */
-			status->link_mode = WIFI_6;
+			status->link_mode = wpa_s->connection_he ? 6 :
+                        (wpa_s->connection_vht ? 5 : 4);
```

Comment:
Can we use macros instead of magic numbers?

Reply:
Also is the above TODO still relevant?

Reply:
FYI, defines are here https://github.com/nrfconnect/sdk-zephyr/blob/main/include/zephyr/net/wifi.h#L202